pkg/resource/secret_reconciler.go

```diff
@@ -57,7 +57,9 @@ const (
 // by propagating their data to another secret. Both secrets must consent to
 // this process by including propagation annotations. The Reconciler assumes it
 // has a watch on both propagating (from) and propagated (to) secrets.
-func NewSecretPropagatingReconciler(m manager.Manager) reconcile.Reconciler {
+func NewSecretPropagatingReconciler(m manager.Manager) reconcile.Reconciler { // nolint:gocyclo
+	// NOTE(hasheddan): This function is over our cyclomatic complexity goal.
+	// Be wary of adding additional complexity.
```

Comment:
I think you removed this linter comment from the other PR - is it sneaking back in here?

Reply:
ah yes, I must have messed up the rebase.